datasharing - Redshift adapter (#8920)
fishtown-analytics/docs.getdbt.com: website/docs/docs/dbt-versions/core-upgrade/04-upgrading-to-v1.11.md

### Redshift

> - The new `datasharing` profile credential migrates metadata queries from PostgreSQL catalog tables (`pg_*`, `information_schema`) to Redshift's native SHOW commands (for example, `SHOW TABLES`, `SHOW COLUMNS`, `SHOW SCHEMAS`). This enables cross-database and cross-cluster access with [Redshift Datasharing](https://docs.aws.amazon.com/redshift/latest/dg/datashare-overview.html). Refer to [Redshift setup](/docs/local/connect-data-platform/redshift-setup) for configuration details.

Author: ```suggestion
- The new `datasharing` profile credential migrates metadata queries from PostgreSQL catalog tables (for example, `pg_*`, `information_schema`) to Redshift's native SHOW commands (for example, `SHOW TABLES`, `SHOW COLUMNS`, `SHOW SCHEMAS`). This enables cross-database and cross-cluster access with [Redshift Datasharing](https://docs.aws.amazon.com/redshift/latest/dg/datashare-overview.html). Refer to [Redshift setup](/docs/local/connect-data-platform/redshift-setup) for configuration details.
```

> - The new `datasharing` profile credential migrates metadata queries from PostgreSQL catalog tables (for example, `pg_*`, `information_schema`) to Redshift's native `SHOW` commands (for example, `SHOW TABLES`, `SHOW COLUMNS`, `SHOW SCHEMAS`). This enables cross-database and cross-cluster access with [Redshift Datasharing](https://docs.aws.amazon.com/redshift/latest/dg/datashare-overview.html). Refer to [Redshift setup](/docs/local/connect-data-platform/redshift-setup) for configuration details.

Reviewer: Hiya Bianca,

I had a look at the issues linked to the PR (https://github.com/dbt-labs/dbt-adapters/issues/637 and https://github.com/dbt-labs/dbt-adapters/issues/1620) and they suggest framing this as adapter-level metadata migration, with the `datasharing` profile credential enabling that path.

So maybe a rewrite like:

"The new `datasharing` profile credential enables `dbt-redshift` to use Redshift-native metadata APIs (including `SVV_* views` and SHOW commands) instead of PostgreSQL catalog tables such as `pg_*` and `information_schema`. This supports cross-database and cross-cluster access with Redshift Datasharing. For setup details, refer to [Redshift setup](link).

Author: Hey @nataliefiann , thanks for the suggestion! What do you think of the following revision? I dropped "API" since these are SQL-level interfaces, and removed `SVV_*` since the implementation uses `SHOW` commands:

```suggestion
- The new `datasharing` profile credential enables `dbt-redshift` to use Redshift-native metadata commands (`SHOW` commands such as `SHOW TABLES` and `SHOW COLUMNS`) instead of PostgreSQL catalog tables such as `pg_*` and `information_schema`. This supports cross-database and cross-cluster access with [Redshift Datasharing](https://docs.aws.amazon.com/redshift/latest/dg/datashare-overview.html). For configuration details, refer to [Redshift setup](/docs/local/connect-data-platform/redshift-setup).
```

@tauhid621 could you also review for accuracy? Thanks!

> - The new `datasharing` profile credential migrates metadata queries from PostgreSQL catalog tables (`pg_*`, `information_schema`) to Redshift's native `SVV_*` system views (for example, `SVV_ALL_TABLES`, `SVV_ALL_COLUMNS`, `SVV_ALL_SCHEMAS`). This enables cross-database and cross-cluster access with [Redshift Datasharing](https://docs.aws.amazon.com/redshift/latest/dg/datashare-overview.html). Refer to [Redshift setup](/docs/local/connect-data-platform/redshift-setup) for configuration details.

Reviewer 2: ```suggestion
- The new `datasharing` profile credential migrates metadata queries from PostgreSQL catalog tables (`pg_*`, `information_schema`) to Redshift's native SHOW commands (for example, `SHOW TABLES`, `SHOW COLUMNS`, `SHOW SCHEMAS`). This enables cross-database and cross-cluster access with [Redshift Datasharing](https://docs.aws.amazon.com/redshift/latest/dg/datashare-overview.html). Refer to [Redshift setup](/docs/local/connect-data-platform/redshift-setup) for configuration details.
```
